### PR TITLE
add CMakeLists.txt to build lib/ bin/ with cmake

### PR DESCRIPTION
**Pull request by YusukeSuzuki**
this commit enables to develop ccv with makefile / xcode / visual studio etc.

### CODE REVIEW

**Comment by liuliu** [comment]
Thanks. This is a much clearer CMake file than most of them. I will take a look at it tonight.

**Comment by patricksnape** [comment]
Although I've never been active here, I've been following the project for a while, particularly because I am very interested in its object detection implementations.

My real use case is that I would like to create a thin wrapper around libccv to be used in Python, and this requires an unfortunate build system whereby I need to build on Linux, Mac and Windows. Specifically, on Windows, I need to be able to have _different_ solutions for Visual Studio 2010 and Visual Studio 2008. Therefore, something like CMake is a godsend for managing the fact I need 4 different build systems.

In short, I just wanted to throw my +1 into the ring for supporting a CMake build system.

**Comment by liquidzym** [comment]
sorry to post it here, i try with your cmake file,but seems not working,the output error is:
CMake Error at ../share/cmake-3.2/Modules/FindPackageHandleStandardArgs.cmake:138 (message):
  Could NOT find PNG (missing: PNG_LIBRARY PNG_PNG_INCLUDE_DIR)
Call Stack (most recent call first):
  ../share/cmake-3.2/Modules/FindPackageHandleStandardArgs.cmake:374 (_FPHSA_FAILURE_MESSAGE)
  ../share/cmake-3.2/Modules/FindPNG.cmake:121 (find_package_handle_standard_args)
  lib/CMakeLists.txt:13 (find_package)
@YusukeSuzuki ,any idea how to fix it? 
cheers

**Comment by YusukeSuzuki** [comment]
@liquidzym Could you tell me your environment (OS, distibution, cmake version)?

**Comment by liquidzym** [comment]
@YusukeSuzuki,i'm using cmake 3.2.1 running under the win8.1 with vs2013 
cheers

**Comment by YusukeSuzuki** [comment]
@liquidzym 
you should install zlib and libpng and maybe libjpeg. 
i build zlib and libpng with cmake-gui on windows.
it need to configure library file path and include path with your hand(PNG_LIBRARY PNG_PNG_INCLUDE_DIR and ZLIB_~~~).
i've installed zlib and libpng, but i've not installed libjpeg yet(libjpeg does'nt have CMakeLists.txt).
sorry, i usually use linux and os x :sweat:
